src/OrleansServiceBus/Providers/Streams/EventHub/ICheckpointSettings.cs

```diff
@@ -0,0 +1,99 @@
+﻿
+using System;
+using System.Collections.Generic;
+using Orleans.Providers;
+
+namespace Orleans.ServiceBus.Providers
+{
+    public interface ICheckpointSettings
+    {
+        /// <summary>
+        /// Azure table storage data connections string
+        /// </summary>
+        string DataConnectionString { get; }
+
+        /// <summary>
+        /// Azure storage table name where the checkpoints will be stored
+        /// </summary>
+        string TableName { get; }
+
+        /// <summary>
+        /// How often to persist the checkpoints, if they've changed.
+        /// </summary>
+        TimeSpan PersistCadence { get; }
+
+        string DeploymentId { get; }
```

Comment:
sure, ServiceId would work. My only worry is that we have another ServiceId configuration property, and might not be clear that they aren't related... Still, I'm not opposed to it, but definititely not DeploymentId :)